test/test_linting.py

```diff
@@ -852,3 +772,34 @@ def test_invalid_identifiers():
                   doi:10.1093/bioinformatics/btr010
         ''']
     )
+
+
+def test_deprecated_numpy_spec():
+    run_lint(
+        func=lint_functions.deprecated_numpy_spec,
+        should_pass=['''
+        a:
+            meta.yaml: |
+              package:
+                name: a
+                version: 0.1
+              requirements:
+                build:
+                  - numpy
+                  - python
+                run:
+                  - numpy
```

Comment:
I think the new way specifically for numpy is:
```
requirements:
        build:
             - numpy
             - python
        run:
             - {{ pin_compatible('numpy') }}
```

Reply:
We use the conda-forge feedstock for the pinning (i.e. pinning at the variant level). Hence, I think this is needed here:

```
requirements:
  host:
    - numpy {{ numpy }}
  run:
    - numpy {{ numpy }}
```

The `pin_compatible` seems to be only needed when overwriting the variants matrix on a per-recipe base.

Reply:
We should be fine with 
```yaml
  host:
    - numpy
```
instead of
```yaml
  host:
    - numpy {{ numpy }}
```
(or rather
```yaml
  host:
    - numpy {{ numpy }}.*
```
if `numpy` is given as a plain version as in `conda-forge-pinning`) since the top-level keys in the `conda_build_config.yaml` files act two-fold: For one, they can be used as arbitrary Jinja2 variables like with `numpy {{ numpy }}.*` where they just get replaced by the given value (for the currently enumerated build variant). Secondly, they are interpreted as versions (or rather Conda match specifications, I think) for packages with the same name in `build`/`host` requirements.

As for `pin_compatible('numpy')`, this is a special case since `numpy` deliberately does not define `run_exports` in https://github.com/AnacondaRecipes/numpy-feedstock/blob/master/recipe/meta.yaml (which I expect `conda-forge` to adapt, too) and also has no `pin_run_as_build` entry. Hence, a plain `numpy` requirement will remain a plain `numpy`.
Although I said earlier that we should avoid `pin_compatible` on a per-recipe basis, with `numpy` we have to add it, though, due to the lacking `run_exports`/`pin_run_as_build`.
(The reason for leaving out `run_exports`/`pin_run_as_build` for `numpy` is that it often does not require the pinning (= if used as a pure Python dependency) and only sometimes requires it (= if a package links to `numpy`'s C-API).)